src/transformers/modeling_bart.py

```diff
@@ -939,8 +939,30 @@ def forward(
         return outputs
 
     @staticmethod
-    def prepare_inputs_for_generation(input_ids, past, **kwargs):
-        return {"input_ids": input_ids, "decoder_cached_states": past, "decoder_input_ids": input_ids}
+    def prepare_inputs_for_generation(decoder_input_ids, past, encoder_inputs, encoder_outputs, **wwargs):
+        # if encoder_outputs are not defined, the decoder input should be none, since it's the first decoding step
+        if encoder_outputs is None:
+            decoder_input_ids = None
+
+        # the input_ids for Bart always correspond to the encoder_inputs, which don't change during decoding
+        inputs_ids = encoder_inputs
+        return {
+            "input_ids": inputs_ids,
+            "encoder_outputs": encoder_outputs,
+            "decoder_cached_states": past,
+            "decoder_input_ids": decoder_input_ids,
+        }
+
+    def postprocess_outputs_for_generation(self, outputs):
```

Comment:
This function in general is very hacky and should be removed when changing to named tuples as the preferred output for all models.